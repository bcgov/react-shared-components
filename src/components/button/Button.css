.bcgov-normal-blue {
  color: white;
  font-weight: 700;
  font-style: normal;
  background-color: #003366;
}

.bcgov-normal-blue:hover {
  background-color: rgba(43, 91, 164, 1);
  color: white;
}

.bcgov-normal-white {
  color: #003366;
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(204, 204, 204, 1);
}

.bcgov-normal-white:hover {
  box-sizing: border-box;
  border-width: 1px;
  border-style: solid;
  background-color: rgba(242, 242, 242, 1);
  border-color: rgba(204, 204, 204, 1);
}

.bcgov-consistent-width {
  width: 270px;
}

.bcgov-button {
  min-width: 150px;
  min-height: 40px;
}

@media (max-width: 575.98px) {
  .bcgov-button {
    min-width: 0px;
    min-height: 0px;
  }
}

.bcgov-loader-show {
  display: inline-block;
}
