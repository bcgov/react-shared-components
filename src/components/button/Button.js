import React from "react";
import PropTypes from "prop-types";
import { Loader } from "../loader/Loader";
import "./Button.css";

export const Button = ({
  hasLoader,
  onClick,
  label,
  styling,
  disabled,
  testId,
}) => (
  <button
    className={`bcgov-button ${styling}`}
    onClick={onClick}
    type="button"
    disabled={disabled}
    data-test-id={testId}
  >
    {label}
    {hasLoader && (
      <div className="bcgov-loader-show">
        <Loader />
      </div>
    )}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func,
  label: PropTypes.string.isRequired,
  styling: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  testId: PropTypes.string,
  hasLoader: PropTypes.bool,
};

Button.defaultProps = {
  disabled: false,
  testId: "",
  hasLoader: false,
  onClick: () => {},
};
