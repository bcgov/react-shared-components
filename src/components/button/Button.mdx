import { Meta, Story } from "@storybook/addon-docs/blocks";
import { Button } from "./Button";

<Meta title="Button" component={Button} />

# Button

Buttons allow users to take actions, and make choices, with a single click.

## API

| Prop      | Type     | Required | Description                                                                           | Default |
| --------- | -------- | -------- | ------------------------------------------------------------------------------------- | ------- |
| onClick   | function | true     | function executed on button click                                                     |         |
| label     | string   | true     | text to be displayed on button                                                        |         |
| styling   | string   | true     | list of css classnames to apply styling based on (ex: normal-blue, normal-white, btn) |         |
| disabled  | boolean  | false    | indicates whether button is disabled or enabled                                       | false   |
| hasLoader | boolean  | false    | indicates whether you want to show a loader within button                             | false   |

<br />

## Primary Button

Primary buttons allow users to carry out an important action on your service, such as Download or Submit.

### When to Use

The most important actions you want users to take on your site.

### When not to Use

Supporting actions on your page such as Cancel, Delete, or supporting information.

<Story name="Primary" id="button--primary"></Story>

```bash
<Button onClick={onClick} label="Submit" styling="normal-blue btn" />
```

## Secondary Button

Secondary buttons allow users to carry out a supporting action on your service, such as Back or Cancel.

### When to Use

Supporting actions users can take on your site.

### When not to Use

Primary actions on your page such as Submit, or Start. Use a `Primary Button` instead.

<Story name="Secondary" id="button--secondary"></Story>

```bash
<Button onClick={onClick} label="Submit" styling="normal-white btn" />
```

## Disabled Button

A button that can’t be clicked and doesn’t react to hover. A disabled button is
greyed out to indicate to users that they cannot undertake the action associated
with it. This is usually for page logic reasons.

<Story name="Disabled" id="button--disabled"></Story>

```bash
<Button onClick={onClick} label="Disabled" styling="normal-blue btn" disabled />
```

## Has Loader

A button that you want to display a loader within, for when an async action is
taking place that you want the user to wait for before proceeding.

<Story name="Has Loader" id="button--has-loader"></Story>

```bash
<Button
  onClick={onClick}
  label="Has Loader"
  styling="normal-blue btn"
  hasLoader
/>
```
