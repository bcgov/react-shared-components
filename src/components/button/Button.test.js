import React from "react";
import testBasicSnapshot from "../../TestHelper";

import { Button } from "./Button";

describe("Button Component", () => {
  const onClick = jest.fn();
  const label = "Button label";

  test("Matches the 'primary' snapshot", () => {
    const buttonComponent = (
      <Button onClick={onClick} label={label} styling="normal-blue btn" />
    );

    testBasicSnapshot(buttonComponent);
  });

  test("Matches the 'secondary' snapshot", () => {
    const buttonComponent = (
      <Button onClick={onClick} label={label} styling="normal-white btn" />
    );

    testBasicSnapshot(buttonComponent);
  });

  test("Matches the 'disabled' snapshot", () => {
    const buttonComponent = (
      <Button
        onClick={onClick}
        label={label}
        styling="normal-blue btn"
        disabled
      />
    );

    testBasicSnapshot(buttonComponent);
  });

  test("Matches the 'hasLoader' snapshot", () => {
    const buttonComponent = (
      <Button
        hasLoader
        onClick={onClick}
        label={label}
        styling="normal-blue btn"
      />
    );

    testBasicSnapshot(buttonComponent);
  });
});
