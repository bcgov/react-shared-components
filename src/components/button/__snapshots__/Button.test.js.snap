// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Matches the 'disabled' snapshot 1`] = `
<button
  className="bcgov-button bcgov-normal-blue btn"
  data-testid=""
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  Button label
</button>
`;

exports[`Button Component Matches the 'hasLoader' snapshot 1`] = `
<button
  className="bcgov-button bcgov-normal-blue btn"
  data-testid=""
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Button label
  <div
    className="bcgov-loader-show"
  >
    <div
      className="bcgov-btn-loader"
    />
  </div>
</button>
`;

exports[`Button Component Matches the 'primary' snapshot 1`] = `
<button
  className="bcgov-button bcgov-normal-blue btn"
  data-testid=""
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Button label
</button>
`;

exports[`Button Component Matches the 'secondary' snapshot 1`] = `
<button
  className="bcgov-button bcgov-normal-white btn"
  data-testid=""
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Button label
</button>
`;
